app/routes/projects/$projectId/comment/$id.tsx

```diff
@@ -0,0 +1,47 @@
+import type { ActionFunction } from "@remix-run/node";
+import { redirect } from "@remix-run/node";
+import invariant from "tiny-invariant";
+import { getSession, requireProfile, requireUser } from "~/session.server";
+import { validator } from "~/core/components/Comments";
+import {
+  deleteComment,
+  getComment,
+  updateComment,
+} from "~/models/comment.server";
+import { adminRoleName } from "~/constants";
+
+export const action: ActionFunction = async ({ request, params }) => {
+  invariant(params.projectId, "projectId could not be found");
+  invariant(params.id, "comment id could not be found");
+  const projectId = params.projectId;
+  const id = params.id;
+
+  try {
+    const user = await requireUser(request);
+    const profile = await requireProfile(request);
+    const isAdmin = user.role == adminRoleName;
+    const comment = await getComment(id);
+
+    if (!isAdmin || comment.authorId != profile.id) {
```

Comment:
If I'm not an admin and I want to delete the message, the same message will be displayed, okay? I mean, we don't need another message when I want to delete?

Reply:
I don't understand the comment 🙈.

The `if` is meant so that only admins or comment authors can edit or delete comments. Otherwise, it will throw an exception.

Reply:
I said it for the message that says _Edit comment not allowed_ , that maybe we can modify it if its an error for the delete action and not for the edit 😛

Reply:
Oh i see! Yes, ok, just changed it to `Edit/Delete comment not allowed`. This message is not displayed to the user though, it is only showed on the server logs. I think this is OK, since on normal usage a user should not be POSTing or DELETEing on that route if they don't have permissions.

Also a future task is adding an application wide snackbar for session flash messages, that I added on this PR but I'm still not displaying anywhere 🙈 . For that I think it we could either use https://mui.com/material-ui/react-snackbar/ or https://github.com/iamhosseindhv/notistack (suggested on the snackbar page :-p).